src/app/(Authentication)/SignUp/Address/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { router, useLocalSearchParams } from 'expo-router';
+import React, { useState } from 'react';
+import { Text, View, Image } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import styles from './styles';
+import Check from '../../../../../assets/check-white.svg';
+import AuthInput from '../../../../Components/AuthInput/AuthInput';
+import { signUpUser } from '../../../../supabase/queries/auth';
+
+export default function SignUpScreen() {
+  const { name } = useLocalSearchParams() as unknown as { name: string };
+  const { email } = useLocalSearchParams() as unknown as { email: string };
+  const { password } = useLocalSearchParams() as unknown as {
+    password: string;
+  };
+  const [streetAddress, setStreetAddress] = useState<string>('');
+  const [city, setCity] = useState<string>('');
+  const [state, setState] = useState<string>('');
+  const [zipcode, setZipcode] = useState<string>('');
+
+  const [errorExists, setErrorExists] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const onChangeStreetAddress = (text: string) => {
+    setErrorExists(false);
+    setStreetAddress(text);
+  };
+
+  const onChangeCity = (text: string) => {
+    setErrorExists(false);
+    setCity(text);
+  };
+
+  const onChangeState = (text: string) => {
+    setErrorExists(false);
+    setState(text);
+  };
+
+  const onChangeZipcode = (text: string) => {
+    setErrorExists(false);
+    setZipcode(text);
+  };
+
+  const validateAddressInputs = () => {
+    // Insert any address input validation logic
+    return true;
+  };
+
+  const handleSubmit = () => {
```

Comment:
I tried to simplify the error-checking flow. Basically, it shouldn't perform any input validation until after a user tries to "submit" or "confirm" depending on the page. That way all error checking can be performed in one function call, and once the first error is detected, it'll indicate that an `errorExists`, then set the `errorMessage` accordingly. As of now, the only pre-validation that occurs is making sure that the inputs are non-empty, and it will disable submission until then.